smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/protocols/ProtocolPriority.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.typescript.codegen.protocols;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import software.amazon.smithy.model.shapes.ShapeId;
+
+
+/**
+ * Allows customization of protocol selection for specific services or a global default ordering.
+ */
+public final class ProtocolPriority {
```

Comment:
Global mutable state it's a bad pattern, error prone and hard to debug, and, as coded not thread safe. Is there a better way to define this without having a global state?

Reply:
turned this into a non-static class and is now instantiated under `TypeScriptSettings`, which is singular in a ProtocolGenerationContext.